Flaky `test_fk()` results

- Fails occasionally due to a small difference in the x-component of the transform matrix
  - https://ci.appveyor.com/project/nnadeau/pybotics/build/1.0.732/job/c9mmdfvctt9jasie
  - https://ci.appveyor.com/project/nnadeau/pybotics/build/1.0.732/job/qs645jqgd49iwa3s
- This corresponds to the first row of the UR10 resources
```
-45.0,147.0,-39.0,96.0,49.0,67.0,-0.10936549564013165,0.9937209495290638,0.02361912001841469,341.25528339185024,-0.30766985716369466,-0.011247226994026319,-0.9514266965341137,-658.5887448195482,-0.       9451869906829335,-0.11132014351410438,0.30696804115695675,-625.3245786240964,0.0,0.0,0.0,1.0
```

Repro case:
- See log from https://travis-ci.org/nnadeau/pybotics/jobs/357802368 
[log.txt](https://github.com/nnadeau/pybotics/files/1844665/log.txt)
- no `pytest-xdist`
- `--randomly-seed=1521905437`

```
$ python setup.py test
INFO:root:README path: /home/travis/build/nnadeau/pybotics/README.md
INFO:root:Requirements path: /home/travis/build/nnadeau/pybotics/requirements.txt
/opt/python/3.4.6/lib/python3.4/distutils/dist.py:260: UserWarning: Unknown distribution option: 'long_description_content_type'
  warnings.warn(msg)
running pytest
running egg_info
writing requirements to pybotics.egg-info/requires.txt
writing top-level names to pybotics.egg-info/top_level.txt
writing dependency_links to pybotics.egg-info/dependency_links.txt
writing pybotics.egg-info/PKG-INFO
reading manifest template 'MANIFEST.in'
writing manifest file 'pybotics.egg-info/SOURCES.txt'
running build_ext
============================= test session starts ==============================
platform linux -- Python 3.4.6, pytest-3.3.2, py-1.5.3, pluggy-0.6.0 -- /home/travis/virtualenv/python3.4.6/bin/python
cachedir: .cache
Using --randomly-seed=1521905437
rootdir: /home/travis/build/nnadeau/pybotics, inifile: pytest.ini
plugins: randomly-1.2.3, cov-2.5.1, hypothesis-3.50.3
collected 34 items                                                             
tests/test_errors.py::test_errors PASSED                                 [  2%]
tests/test_tool.py::test_tool PASSED                                     [  5%]
tests/test_predefined_models.py::test_models PASSED                      [  8%]
tests/test_link.py::test_to_dict PASSED                                  [ 11%]
tests/test_link.py::test_repr PASSED                                     [ 14%]
tests/test_link.py::test_len PASSED                                      [ 17%]
tests/test_link.py::test_displace PASSED                                 [ 20%]
tests/test_validation.py::test_is_4x4_matrix PASSED                      [ 23%]
tests/test_validation.py::test_is_vector PASSED                          [ 26%]
tests/test_kinematic_chain.py::test_init PASSED                          [ 29%]
tests/test_kinematic_chain.py::test_to_dict PASSED                       [ 32%]
tests/test_kinematic_chain.py::test_num_parameters PASSED                [ 35%]
tests/test_kinematic_chain.py::test_repr PASSED                          [ 38%]
tests/test_kinematic_chain.py::test_vector PASSED                        [ 41%]
tests/test_optimization.py::test_compute_absolute_errors PASSED          [ 44%]
tests/test_optimization.py::test_optimization PASSED                     [ 47%]
tests/test_optimization.py::test_handler_validate_transform_mask PASSED  [ 50%]
tests/test_geometry.py::test_translation_matrix PASSED                   [ 52%]
tests/test_geometry.py::test_vector_2_matrix PASSED                      [ 55%]
tests/test_geometry.py::test_orientation PASSED                          [ 58%]
tests/test_geometry.py::test_wrap_2_pi PASSED                            [ 61%]
tests/test_geometry.py::test_matrix_2_vector PASSED                      [ 64%]
tests/test_geometry.py::test_rotation_matrix_xyz PASSED                  [ 67%]
tests/test_robot.py::test_joint_limits PASSED                            [ 70%]
tests/test_robot.py::test_fk FAILED                                      [ 73%]
tests/test_robot.py::test_jacobian_flange PASSED                         [ 76%]
tests/test_robot.py::test_ik PASSED                                      [ 79%]
tests/test_robot.py::test_repr PASSED                                    [ 82%]
tests/test_robot.py::test_jacobian_world PASSED                          [ 85%]
tests/test_robot.py::test_random_joints PASSED                           [ 88%]
tests/test_robot.py::test_str PASSED                                     [ 91%]
tests/test_robot.py::test_home_position PASSED                           [ 94%]
tests/test_robot.py::test_len PASSED                                     [ 97%]
tests/test_robot.py::test_compute_joint_torques PASSED                   [100%]
----------- coverage: platform linux, python 3.4.6-final-0 -----------
Name                            Stmts   Miss Branch BrPart  Cover   Missing
---------------------------------------------------------------------------
pybotics/__init__.py                9      0      0      0   100%
pybotics/constants.py               5      0      0      0   100%
pybotics/errors.py                  9      0      2      0   100%
pybotics/geometry.py               90      0      8      0   100%
pybotics/json_encoder.py            4      0      0      0   100%
pybotics/kinematic_chain.py        72      0     12      0   100%
pybotics/link.py                   67      0      0      0   100%
pybotics/optimization.py           57      0      8      0   100%
pybotics/predefined_models.py      20      0      0      0   100%
pybotics/robot.py                 109      0     14      0   100%
pybotics/tool.py                   41      0      4      0   100%
pybotics/validation.py             15      0      8      0   100%
---------------------------------------------------------------------------
TOTAL                             498      0     56      0   100%
Required test coverage of 100% reached. Total coverage: 100.00%
=================================== FAILURES ===================================
___________________________________ test_fk ____________________________________
resources_path = PosixPath('/home/travis/build/nnadeau/pybotics/tests/resources')
    def test_fk(resources_path: Path):
        """
        Test robot.
    
        :param robot:
        :return:
        """
        # get resource
        path = resources_path / 'ur10-joints-poses.csv'
        data = np.loadtxt(str(path), delimiter=',')
        if data.ndim == 1:
            data = np.expand_dims(data, axis=0)
    
        # load robot
        robot = UR10()
    
        # test
        for d in data:
            joints = np.deg2rad(d[:robot.ndof])
            desired_pose = d[robot.ndof:].reshape(TRANSFORM_MATRIX_SHAPE)
    
            atol = 1e-3
    
            # test with position argument
            actual_pose = robot.fk(q=joints)
>           np.testing.assert_allclose(actual_pose, desired_pose, atol=atol)
E           AssertionError: 
E           Not equal to tolerance rtol=1e-07, atol=0.001
E           
E           (mismatch 6.25%)
E            x: array([[ -1.093655e-01,   9.937209e-01,   2.361912e-02,   3.413553e+02],
E                  [ -3.076699e-01,  -1.124723e-02,  -9.514267e-01,  -6.585887e+02],
E                  [ -9.451870e-01,  -1.113201e-01,   3.069680e-01,  -6.253246e+02],
E                  [  0.000000e+00,   0.000000e+00,   0.000000e+00,   1.000000e+00]])
E            y: array([[ -1.093655e-01,   9.937209e-01,   2.361912e-02,   3.412553e+02],
E                  [ -3.076699e-01,  -1.124723e-02,  -9.514267e-01,  -6.585887e+02],
E                  [ -9.451870e-01,  -1.113201e-01,   3.069680e-01,  -6.253246e+02],
E                  [  0.000000e+00,   0.000000e+00,   0.000000e+00,   1.000000e+00]])
tests/test_robot.py:42: AssertionError
===================== 1 failed, 33 passed in 17.54 seconds =====================
```

--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -18,7 +18,6 @@ pytest==3.3.2
 pytest-cov==2.5.1
 pytest-randomly==1.2.3
 pytest-runner==4.2
-pytest-xdist==1.22.2
 readme_renderer==17.4
 scikit-learn==0.19.1
 scipy==1.0.0

--- a/pybotics/predefined_models.py
+++ b/pybotics/predefined_models.py
@@ -1,4 +1,5 @@
 """Predefined robot models."""
+from copy import deepcopy
 from typing import Any
 
 import numpy as np  # type: ignore
@@ -25,7 +26,7 @@ class KukaLBRiiwa7(Robot):
 
     def __init__(self, **kwargs: Any) -> None:
         """Init robot."""
-        super().__init__(self.kinematic_chain, **kwargs)
+        super().__init__(deepcopy(self.kinematic_chain), **kwargs)
 
 
 class MecademicMeca500(Robot):
@@ -45,7 +46,7 @@ class MecademicMeca500(Robot):
 
     def __init__(self, **kwargs: Any) -> None:
         """Init robot."""
-        super().__init__(self.kinematic_chain, **kwargs)
+        super().__init__(deepcopy(self.kinematic_chain), **kwargs)
 
 
 class PUMA560(Robot):
@@ -65,7 +66,7 @@ class PUMA560(Robot):
 
     def __init__(self, **kwargs: Any) -> None:
         """Init robot."""
-        super().__init__(self.kinematic_chain, **kwargs)
+        super().__init__(deepcopy(self.kinematic_chain), **kwargs)
 
 
 class UR10(Robot):
@@ -85,4 +86,4 @@ class UR10(Robot):
 
     def __init__(self, **kwargs: Any) -> None:
         """Init robot."""
-        super().__init__(self.kinematic_chain, **kwargs)
+        super().__init__(deepcopy(self.kinematic_chain), **kwargs)

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,2 +1,2 @@
 [pytest]
-addopts = --verbose --cov=pybotics --cov-report term-missing --cov-config .coveragerc -nauto
+addopts = --verbose --cov=pybotics --cov-report term-missing --cov-config .coveragerc

